Sources/PopupBridge/POPPopupBridge.swift

```diff
@@ -1,63 +1,61 @@
 import Foundation
-import SafariServices
 import WebKit
+import AuthenticationServices
 
 @objcMembers
-public class POPPopupBridge: NSObject, WKScriptMessageHandler, SFSafariViewControllerDelegate {
+public class POPPopupBridge: NSObject, WKScriptMessageHandler {
+
+    /// Exposed for testing
+    var returnedWithURL: Bool = false
     
     // MARK: - Private Properties
     
     private let messageHandlerName = "POPPopupBridge"
-    private let hostName = "popupbridgev1"
-    
+    private let hostName = "popupbridgev1"    
     private let webView: WKWebView
-    private let urlScheme: String
-    private let delegate: POPPopupBridgeDelegate
-    private var safariViewController: SFSafariViewController?
+    private var webAuthenticationSession: WebAuthenticationSession
     
-    private static var returnBlock: ((URL) -> Bool)?
+    private var returnBlock: ((URL) -> Void)? = nil
     
-    // MARK: - Initializer
+    // MARK: - Initializers
         
     /// Initialize a Popup Bridge.
     /// - Parameters:
     ///   - webView: The web view to add a script message handler to. Do not change the web view's configuration or user content controller after initializing Popup Bridge.
-    ///   - urlScheme: The URL Scheme that you have registered in your Info.plist.
-    ///   - delegate: A delegate that presents and dismisses the pop-up (a SFSafariViewController).
-    public init(
-        webView: WKWebView,
-        urlScheme: String,
-        delegate: POPPopupBridgeDelegate
-    ) {
+    public init(webView: WKWebView) {
         self.webView = webView
-        self.urlScheme = urlScheme
-        self.delegate = delegate
+        self.webAuthenticationSession = WebAuthenticationSession()
         
         super.init()
-        
-        configureWebView(scheme: urlScheme)
-        
-        POPPopupBridge.returnBlock = { (url : URL) -> Bool in
+
+        configureWebView(scheme: PopupBridgeConstants.callbackURLScheme)
+                
+        returnBlock = { url in
             guard let script = self.constructJavaScriptCompletionResult(returnURL: url) else {
-                return false
+                return
             }
-            
+
             self.injectWebView(webView: webView, withJavaScript: script)
-            return true
+            return
         }
     }
-    
-    // MARK: - Public Methods
-    
-    /// Handle completion of the popup flow by calling this method from either
-    /// your scene:openURLContexts: scene delegate method or
-    /// your application:openURL:sourceApplication:annotation: app delegate method.
-    @objc(openURL:)
-    public static func open(url: URL) -> Bool {
-        if let returnBlock {
-            return returnBlock(url)
-        } else {
-            return false
+
+    /// Exposed for testing
+    init(webView: WKWebView, webAuthenticationSession: WebAuthenticationSession) {
```

Comment:
Ideally we avoid having duplicate logic just for dependency injection for tests. This opens the door for errors since we have to maintain 2 inits to be the exact same and the real one isn't actually being tested.

We could do something like this:
```
    // MARK: - Private Properties
    
    private let webView: WKWebView
    private var webAuthenticationSession: WebAuthenticationSession = WebAuthenticationSession()
    ...
    
    // MARK: - Initializers
        
    /// Merchant facing init
    public init(webView: WKWebView) {
        self.webView = webView
        
        super.init()

        configureWebView(scheme: PopupBridgeConstants.callbackURLScheme)
                
        returnBlock = { url in
            guard let script = self.constructJavaScriptCompletionResult(returnURL: url) else {
                return
            }

            self.injectWebView(webView: webView, withJavaScript: script)
            return
        }
    }

    /// Exposed for testing
    convenience init(webView: WKWebView, webAuthenticationSession: WebAuthenticationSession) {
        self.init(webView: webView)
        self.webAuthenticationSession = webAuthenticationSession
    }
```

Reply:
Or, not have a 2nd init at all and just expose the `webAuthSession` property. 

```
    // MARK: - Internal Properties
    
    /// Exposed for testing
    var webAuthenticationSession: WebAuthenticationSession = WebAuthenticationSession()

    // MARK: - Initializers
        
    /// Merchant facing init
    public init(webView: WKWebView) {
        self.webView = webView
        
        super.init()

        configureWebView(scheme: PopupBridgeConstants.callbackURLScheme)
                
        returnBlock = { url in
            guard let script = self.constructJavaScriptCompletionResult(returnURL: url) else {
                return
            }

            self.injectWebView(webView: webView, withJavaScript: script)
            return
        }
    }
```

I think the 2nd init is a _little_ more clear and less error prone in the test suite, but I could go either way. What do you think?

Reply:
Yeah I am down for this approach to reduce the duplication!

Reply:
✅ : https://github.com/braintree/popup-bridge-ios/pull/57/commits/1391a3bd090a2b0902dc5d78c7f123f13fd5478d